worker/src/digest_stream.rs

```diff
@@ -0,0 +1,66 @@
+use wasm_bindgen::{self, prelude::*};
+use wasm_bindgen_futures;
+use web_sys::js_sys::{ArrayBuffer, Uint8Array};
+use web_sys::WritableStream;
+
+impl DigestStream {
+    #[allow(dead_code)]
+    pub fn new(algo: DigestStreamAlgorithm) -> Self {
+        Self::internal_new(algo.as_str())
+    }
+
+    #[allow(dead_code)]
+    pub async fn digest(&self) -> Result<Uint8Array, JsValue> {
+        let buffer: ArrayBuffer = self.internal_digest().await?.unchecked_into();
+        Ok(Uint8Array::new(&buffer))
+    }
+}
+
+// from https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest#syntax
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum DigestStreamAlgorithm {
+    Sha1,
+    Sha256,
+    Sha384,
+    Sha512,
+}
+
+#[wasm_bindgen]
+extern "C" {
```

Comment:
This would typically go in `worker-sys`

Reply:
ah, I think I see the pattern now... okay, will fix it up

Reply:
done in https://github.com/cloudflare/workers-rs/commit/2bfa37e50425dc92d2af2ced66a1ecf7f2eab944